### ISSUE TITLE
Several key sequences broken in vi mode

### ISSUE THREAD

**Issue by jurf** [zimfw/zimfw#118]
_(my setup is a bit complicated - I initialize zim with zplug because I need several other plugins and stuff, hence zmanage doesn't work and my gists are a bit long so I won't be posting them here, but that shouldn't matter anyway)_

Description
-----------
Several shortcuts don't have an corresponding vi mode bind, so keys that generate an escape sequnce (like Home/End) end up exiting the input mode, which is confusing, to say at least.

There are other affected binds as well but I'm writing this issue after a long time and forgot about most of them, I'll keep you posted if I find anything more.

Steps to reproduce
------------------
Open terminal, write something, press `Home` and instead of going to the start of the line it will exit the input mode.

**Comment by DoctorJellyface** [comment]
Do you have vi mode activated?

**Comment by DoctorJellyface** [comment]
Hm, when I try `bindkey -e` t Home/End still doesn't work, so it might be something deeper. Just for reference this is Arch on a Thinkpad X220.

**Comment by Tmplt** [comment]
`-e` is for emacs keybinds. With those mappings, Home enters a `~` instead on my keyboard layout (swedish).

**Comment by DoctorJellyface** [comment]
Well, according to [this](https://github.com/Eriner/zim/blob/1f0ed8493872efa1722533e4b43428cf3e75b2f1/modules/input/init.zsh#L57) line, it should not.

**Comment by Eriner** [comment]
This does seem to be a bug.

**Comment by qifei9** [comment]
I have a similar setup and the same issue of home/end keys.

By putting these lines from http://zshwiki.org/home/zle/bindkeys#reading_terminfo into my `.zshrc`, I fixed it.

```
# create a zkbd compatible hash;
# to add other keys to this hash, see: man 5 terminfo
typeset -A key

key[Home]="$terminfo[khome]"
key[End]="$terminfo[kend]"
key[Insert]="$terminfo[kich1]"
key[Backspace]="$terminfo[kbs]"
key[Delete]="$terminfo[kdch1]"
#key[Up]="$terminfo[kcuu1]"
#key[Down]="$terminfo[kcud1]"
key[Left]="$terminfo[kcub1]"
key[Right]="$terminfo[kcuf1]"
key[PageUp]="$terminfo[kpp]"
key[PageDown]="$terminfo[knp]"

# setup key accordingly
[[ -n "$key[Home]"      ]] && bindkey -- "$key[Home]"      beginning-of-line
[[ -n "$key[End]"       ]] && bindkey -- "$key[End]"       end-of-line
[[ -n "$key[Insert]"    ]] && bindkey -- "$key[Insert]"    overwrite-mode
[[ -n "$key[Backspace]" ]] && bindkey -- "$key[Backspace]" backward-delete-char
[[ -n "$key[Delete]"    ]] && bindkey -- "$key[Delete]"    delete-char
#[[ -n "$key[Up]"        ]] && bindkey -- "$key[Up]"        up-line-or-history
#[[ -n "$key[Down]"      ]] && bindkey -- "$key[Down]"      down-line-or-history
[[ -n "$key[Left]"      ]] && bindkey -- "$key[Left]"      backward-char
[[ -n "$key[Right]"     ]] && bindkey -- "$key[Right]"     forward-char

# Finally, make sure the terminal is in application mode, when zle is
# active. Only then are the values from $terminfo valid.
if (( ${+terminfo[smkx]} )) && (( ${+terminfo[rmkx]} )); then
    function zle-line-init () {
        echoti smkx
    }
    function zle-line-finish () {
        echoti rmkx
    }
    zle -N zle-line-init
    zle -N zle-line-finish
fi
```

**Comment by qifei9** [comment]
Enabling the `input` module could also fix this for me. However, currently enabling the `input` module will lead to another issue #134.

**Comment by ericbn** [comment]
As I can see, `bindkey -v` will reset all the previously set keybindings, so it should be executed before any other `bindkey` command (or let's say before loading the `input` module).

**Comment by ericbn** [comment]
This should have been solved by #256 and #261.
